include/can/simlib/socket_transport.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "can/core/message_core.hpp"
+#include "common/core/logging.hpp"
+#include "common/core/synchronization.hpp"
+#include "transport.hpp"
+
+namespace socket_transport {
+
+template <synchronization::LockableProtocol CriticalSection>
+class SocketTransport : public can_transport::BusTransportBase {
+  public:
+    SocketTransport() = default;
+    ~SocketTransport() = default;
+    SocketTransport(const SocketTransport &) = delete;
+    SocketTransport(const SocketTransport &&) = delete;
+    SocketTransport &operator=(const SocketTransport &) = delete;
+    SocketTransport &&operator=(const SocketTransport &&) = delete;
+
+    auto open(const char *ip, uint32_t port) -> bool;
+    void close();
+
+    auto write(uint32_t arb_id, const uint8_t *buff, uint32_t buff_len) -> bool;
+    auto read(uint32_t &arb_id, uint8_t *buff, uint32_t &buff_len) -> bool;
+
+  private:
+    int handle{0};
+    CriticalSection critical_section{};
+};
+
+template <synchronization::LockableProtocol CriticalSection>
+auto SocketTransport<CriticalSection>::open(const char *ip, uint32_t port)
+    -> bool {
+    struct sockaddr_in addr;
+    int s = 0;
+
+    if ((s = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        return false;
+    }
+
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(port);
+    ::inet_aton(ip, &addr.sin_addr);
+
+    if (connect(s, (struct sockaddr *)&addr, sizeof(addr)) == -1) {
```

Comment:
direct bsd socket api... like settling into a warm bath...

Reply:
but also maybe boost sockets? don't really have a preference but it might be easier, allow `unix://` vs `tcp://` protocol specification, allow namelookup, etc

Reply:
I would like to avoid boost until the benefits truly outweigh the pain